src/python/pants/option/parser.py

```diff
@@ -138,6 +138,15 @@ def scope_info(self) -> ScopeInfo:
     def scope(self) -> str:
         return self._scope
 
+    @property
+    def known_args(self) -> frozenset[str]:
+        return frozenset(self._known_args)
+
+    @property
+    def known_scoped_args(self) -> frozenset[str]:
+        prefix = f"{self.scope}-" if self.scope != GLOBAL_SCOPE else ""
```

Comment:
One more nit. When possible, it's more readable to not use negative logic

```suggestion
        prefix = "" if self.scope == GLOBAL_SCOPE else f"{self.scope}-"
```

Reply:
Normally agree, but here I find it more readable to have the "normal" case closer so you first see
```python
prefix = f"{self.scope}-"
```

and then the `if else` that follows is for the exception for the global scope.

Reply:
Yeah, pretty much same reasoning. I understand the point @thejcannon but given that there is a singular scope where we want `""` as the value and nowhere else, so it really is an exception.